indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/SinglePhaseParallelIndexTaskRunner.java

```diff
@@ -111,6 +111,11 @@
   @Override
   public TaskState run() throws Exception
   {
+    if (baseFirehoseFactory.getNumSplits() == 0) {
```

Comment:
If it's 1, the native parallel task currently does the same thing: the supervisor task will run a sub task for the single split. Do you think it's better to process in the supervisor task rather than processing in the sub task?

Reply:
may be i was thinking in that direction, but I guess it's okay to run a sub task for single split.

Reply:
Thanks. I think it makes sense, but probably better to fix in a separate PR.